caf-logging-log4j2/src/main/java/com/github/cafapi/logging/log4j2/ProcessAndThreadIdPatternConverter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.logging.log4j2;
+
+import com.github.cafapi.logging.common.LogMessageValidator;
+import com.hpe.caf.util.processidentifier.ProcessIdentifier;
+import java.util.Locale;
+
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.pattern.ConverterKeys;
+import org.apache.logging.log4j.core.pattern.LogEventPatternConverter;
+import org.apache.logging.log4j.core.pattern.PatternConverter;
+
+@Plugin(name = "ProcessAndThreadIdPatternConverter", category = PatternConverter.CATEGORY)
+@ConverterKeys({ "aProcessId" })
+public final class ProcessAndThreadIdPatternConverter extends LogEventPatternConverter
```

Comment:
Looking at the implementation of this class I feel like there should be a `ProcessAndThreadIdProvider` class in the common module to reduce the duplication.

Reply:
Added a ProcessAndThreadIdProvider